packages/hint-sri/src/hint.ts

```diff
@@ -296,17 +362,130 @@ Actual:   ${hashes.join(', ')}`;
         debug(`Validating integrity of: ${evt.resource}`);
 
         await everySeries(validations, async (validation: Function) => {
-            return await validation(evt);
+            return await validation(evt, origin, cacheErrors);
         });
     }
 
+    private async validateElement(evt: ElementFound, origin: string) {
+        if (this.isInCache(evt)) {
```

Comment:
I think we should try to follow the code flow we had earlier with `validateResource` where we have all the checks in `validations` and split `validateElement` into multiple separate checks (and have a single `validate` method).

E.g.:
```ts
private async isScanEnded(evt, origin) {
            /*
             * The item is cached. For the VSCode extension and the
             * local connector with option 'watch' activated we
             * should report what we have in the cache after the
             * first 'scan::end'.
             */
            if (this.scanEnded) {
                const promises = this.getCache(evt).map((error) => {
                    this.context.report(error.resource, error.message, error.options);
                });

                 await Promise.all(promises);

                 return true;  
            }

             return false;
}

private async isLocalResource(evt, origin) {
        const { resource, element } = evt;
        if (resource.startsWith('file://')) {
            debug(`Ignoring local resource: ${resource}`);

             return Promise.resolve(true);
        }
        return Promise.resolve(false);
}

// And add this checks to `validations` and use the same entry point for all the event handlers
... 
```

Reply:
The think is that if we move all of this to the validation flow, how and when do you do the request to get the content of the element?

It would be weird to have a step in the validation that just download something.